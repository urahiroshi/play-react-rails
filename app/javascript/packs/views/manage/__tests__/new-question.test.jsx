import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewQuestion from '../new-question.jsx'

describe('#render', () => {
  it('should show initial view', () => {
    const component = renderer.create(
      <NewQuestion commands={{}} addQuestion={() => {}} />,
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('when "New Question" button clicked', () => {
  it('should show editing view', () => {
    const wrapper = shallow(
      <NewQuestion commands={{}} addQuestion={() => {}} />,
    )
    wrapper.find('button').simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('when "Cancel" button clicked', () => {
  it('should show initial view', () => {
    const wrapper = shallow(
      <NewQuestion commands={{}} addQuestion={() => {}} />,
    )
    wrapper.find('button').simulate('click')
    wrapper.find('EditingButtons').props().onClickCancel()
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
