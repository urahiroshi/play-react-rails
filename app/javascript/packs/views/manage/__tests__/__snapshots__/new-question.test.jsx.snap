// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render should show initial view 1`] = `
<div
  className="manage-new-question"
>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    New Question
  </button>
</div>
`;

exports[`when "Cancel" button clicked should show initial view 1`] = `
<div
  className="manage-new-question"
>
  <h5>
    New Question:
  </h5>
  <div
    className="col-md-1"
  />
  <div
    className="col-md-6"
  >
    <ContentInput
      defaultValue=""
      disabled={null}
      error=""
      onChange={[Function]}
    />
  </div>
  <div
    className="col-md-3"
  >
    <AnswerInput
      defaultValue=""
      disabled={null}
      error=""
      onChange={[Function]}
    />
  </div>
  <div
    className="col-md-2"
  >
    <EditingButtons
      disabled={null}
      onClickCancel={[Function]}
      onClickSend={[Function]}
    />
  </div>
</div>
`;

exports[`when "New Question" button clicked should show editing view 1`] = `
<div
  className="manage-new-question"
>
  <h5>
    New Question:
  </h5>
  <div
    className="col-md-1"
  />
  <div
    className="col-md-6"
  >
    <ContentInput
      defaultValue=""
      disabled={null}
      error=""
      onChange={[Function]}
    />
  </div>
  <div
    className="col-md-3"
  >
    <AnswerInput
      defaultValue=""
      disabled={null}
      error=""
      onChange={[Function]}
    />
  </div>
  <div
    className="col-md-2"
  >
    <EditingButtons
      disabled={null}
      onClickCancel={[Function]}
      onClickSend={[Function]}
    />
  </div>
</div>
`;
